### PR TITLE
Add human-friendly api.Request printer

### PR DESCRIPTION
**Pull request by mfojtik**
Produces output:

```console
$ sti build --loglevel=2 -e "FOO=BAR,TEST=BAR" foo bar test 

Base Image:		bar
Source:			foo
Output Image Tag:	test
Environment:		FOO=BAR,TEST=BAR
Incremental Build:	disabled
Remove Old Build:	disabled
Force Pull:		enabled
Quiet:			disabled
Layered Build:		disabled
Docker Endpoint:	unix:///var/run/docker.sock
Docker Pull Config:	/home/mfojtik/.dockercfg
Docker Pull User:	mfojtik
```

(the fields are actually aligned, GH messed this up)

### CODE REVIEW

File: pkg/api/printer.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	return out
+}
+
+func printEnv(out io.Writer, env map[string]string) {
+	result := []string{}
+	for k, v := range env {
+		result = append(result, fmt.Sprintf("%s=%s", k, v))
+	}
+	if len(result) == 0 {
```
Comment (soltysh) [inline pkg/api/printer.go]: I'd move that to the beginning and check env length acutally, fail fast, right?

File: pkg/api/printer.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+)
+
+func (r *Request) PrintObj() string {
+	out, err := tabbedString(func(out io.Writer) error {
+		fmt.Fprintf(out, "Base Image:\t%s\n", r.BaseImage)
```
Comment (soltysh) [inline pkg/api/printer.go]: In all our docs we call that the builder image, I'm aware of the var naming, still I'd say to keep with the Builder name.

**Comment by soltysh** [comment]
Two small nits, but LGTM.

File: pkg/api/printer.go (hunk 2 above)
Comment (bparees) [inline pkg/api/printer.go]: +1

File: pkg/api/printer.go (hunk 1 above)
Comment (bparees) [inline pkg/api/printer.go]: +1

**Comment by bparees** [comment]
agree w/ @soltysh's comments and otherwise lgtm.

**Comment by mfojtik** [comment]
Thanks guys, all comments fixed. [merge]

**Pull request merged**